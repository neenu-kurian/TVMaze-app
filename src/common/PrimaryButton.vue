<template>
  <button :type="type" class="btn--primary uppercase" :name="name" @click="handleClick">
    <slot>Search</slot>
  </button>
</template>

<script>
export default {
  name: 'PrimaryButton',
  props: {
    type: String,
    name: String
  },
  methods: {
    // function to go back to homepage/previous page when back button is clicked
    handleClick () {
      if (this.name === 'back') {
        this.$router.push('/show')
      }
    }
  }
}
</script>
