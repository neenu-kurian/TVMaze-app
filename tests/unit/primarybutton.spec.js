import { mount } from '@vue/test-utils'
import PrimaryButton from '@/common/PrimaryButton'

it('displays button text based on slot text', () => {
  const wrapper = mount(PrimaryButton, {
    slots: {
      default: 'Search'
    }
  })

  expect(wrapper.text()).toContain('Search')
})

it('renders the component correctly', () => {
  const wrapper = mount(PrimaryButton, {
    propsData: {
      type: 'submit'
    }
  })

  expect(wrapper).toMatchSnapshot()
})
