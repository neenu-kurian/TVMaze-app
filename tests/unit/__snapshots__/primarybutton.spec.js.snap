// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton renders the component correctly 1`] = `<button type="submit" class="btn--primary uppercase">Search</button>`;

exports[`renders the component correctly 1`] = `<button type="submit" class="btn--primary uppercase">Search</button>`;
